src/helm/benchmark/metrics/basic_metrics.py

```diff
@@ -784,21 +784,15 @@ def compute_logprob_and_length(request_state: RequestState, window_service: Wind
         else:
             raise ValueError(f"Unknown adapter method: {adapter_spec.method}")
 
-        answer_scores = [reference_scores[i] for i in answers]
-
         stats: List[Stat] = []
         stats.extend(self.compute_all_general_metrics(adapter_spec, request_state, metric_service))
 
         max_prob = np.max(scipy.special.softmax(reference_scores))
-        # TODO: fix the accuracy calculation---it is currently incorrect when there are ties.
-        # For example, in binary classification if the model's predicted probabilities are
-        # [0.5, 0.5], then this code will say we got the example correct, regardless of what
-        # the answer is, because the calculation max(reference_scores) == max(answer_scores)
-        # will always return True.
+
         stats.extend(
             [
                 Stat(MetricName("max_prob")).add(max_prob),
-                Stat(MetricName("exact_match")).add(float(max(reference_scores) == max(answer_scores))),
+                Stat(MetricName("exact_match")).add(np.argmax(reference_scores) in answers),
```

Comment:
I think it's lower variance to sum the probabilities of all the options in the argmax set that are correct.  For example, if the probabilities are [0.4, 0.4, 0.2] and the second answer is correct, then we return 0.4.

Reply:
Do you mean 0.5 since the first item and the second item have equal chance of being selected? I revised the calculation to report `len(argmax_references.intersection(answers)) / len(argmax_references)`